config/config_test.go

```diff
@@ -32,8 +35,15 @@ func testConfigType(t reflect.Type) error {
 		return nil
 	case k == reflect.Ptr:
 		pK := t.Elem().Kind()
-		switch {
-		case util.IsPrimitive(pK):
+		if util.IsPrimitive(pK) {
+			return nil
+		}
+		switch t.Elem() {
+		case reflect.TypeOf(v3_2.Clevis{}), reflect.TypeOf(v3_2.Custom{}), reflect.TypeOf(v3_3.Clevis{}), reflect.TypeOf(v3_3.Custom{}):
+			// these structs ended up with pointers; can't be helped now
```

Comment:
are we going to change the structs in 3.3-exp and going forward?

Reply:
I'd like to; see https://github.com/coreos/ignition/issues/1132.  We'd need to discuss it though.